scheduler/src/managers/node_agent/manager.rs

```diff
@@ -22,16 +23,23 @@ impl NodeAgentManager {
         }
     }
 
+    /// Get all agents informations
+    pub fn get_agents(self) -> HashMap<String, NodeAgent> {
+        return self.agents;
+    }
```

Comment:
In order to add consistency, the function needs to take input of a `&self`, else this will borrow the manager and make it unusable later on in the calling function. I went the Iterator route, although you might want to do it another way. The suggestion below is the one used while implementing the suggestion for the `schedule` function.

```suggestion
    pub fn get_agents(&self) -> Iter<'_, String, NodeAgent> {
        return self.agents.iter();
    }
```

Reply:
I think i understand but what is this interest ? Security purposes ?

Reply:
Nope, it's for the purpose of abstraction. The outside world doesn't need to care about what collection you use (in this case, a `HashMap`) to store the agents : you only need to give them what they need, which is a way to iterate through the agents you have stored.
That way, if you need to change this collection type in the future, you won't need to change this function signature, which means you don't change this exposed interface, which finally means that you won't break outside code.